src/components/app.jsx

```diff
@@ -57,9 +60,9 @@ class App extends Component {
     return (
         <div>
           <ChatContainer
-            handleChangeAuthor={this.handleChangeAuthor.bind(this)}
-            handleChangeText={this.handleChangeText.bind(this)}
-            sendMessage={this.sendMessage.bind(this)}
```

Comment:
Don't bind the this context in the render function as that will rebind  the function on every rerender.